codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/shapevisitor/SmithyToDafnyShapeVisitor.java

```diff
@@ -455,7 +455,46 @@ public String doubleShape(DoubleShape shape) {
 
     @Override
     public String unionShape(UnionShape shape) {
-        return "Wrappers.Companion_Option_.Create_None_()";
+        String internalDafnyType = DafnyNameResolver.getDafnyType(shape, context.symbolProvider().toSymbol(shape));
+        writer.addImportFromModule("github.com/dafny-lang/DafnyRuntimeGo", "dafny");
+        String returnString = """
+            func() Wrappers.Option {
+                switch %s.(type) {""".formatted(dataSource);
+        for (var member : shape.getAllMembers().values()) {
+            String memberName = context.symbolProvider().toMemberName(member);
+            String createMemberFunction = memberName.replace(shape.getId().getName() + "Member", "Create_")+"_";
```

Comment:
Use `DafnyNameResolver` for this.

Reply:
`DafnyNameResolver.getDafnyCompanionTypeCreate` is definitely a right choice to use but it returns `Create_MyUnion_` whereas we need member shape of union`Create_IntegerValue_` or `Create_StringValue_` ... 

I did not update `DafnyNameResolver` for this because this only used for union and I don't see any use of this outside union.

Reply:
Also, its unlikely that we will iterate all the members in `DafnyNameResolver`. We can send each member shapes but it will return something like `Create_int32`

Reply:
Create a new function - it most likely will be re-used. Case being aws-sdk.

Reply:
Done at `DafnyNameResolver.getDafnyCreateFuncForUnionMemberShape`